packages/flutter/lib/src/material/overscroll_indicator.dart

```diff
@@ -166,6 +171,25 @@ class _OverscrollIndicatorState extends State<OverscrollIndicator> {
     _hide();
   }
 
+  void _refreshHideTimer() {
+    _hideTimer?.cancel();
+    _hideTimer = new Timer(_kIndicatorTimeoutDuration, _hide);
+  }
+
+  bool _isOverscrolling(double scrollOffset) {
+    return (scrollOffset < _minScrollOffset || scrollOffset > _maxScrollOffset) &&
+      ((scrollOffset - _scrollOffset).abs() > kPixelScrollTolerance.distance);
+  }
+
+  bool _isScrollingInOpposite(double scrollOffset) {
```

Comment:
OK, since I crabbed about the last method: I would have called this _isReverseScroll().